CONTRIBUTING.md

```diff
@@ -1,64 +1,101 @@
-     followed by `git commit -a --amend` to squash any formatting changes
-     into your commit.
-  1. Run `git push origin HEAD:refs/for/master` to upload the review to
-     [Cobalt's Gerrit instance](https://cobalt-review.googlesource.com/).
-  1. Someone from the maintainers team will review the code, putting up comments
-     on any things that need to change for submission.
-  1. If you need to make changes, make them locally, test them, then `git commit
-     --amend` to add them to the *existing* commit. Then return to step 2.
-  1. If you do not need to make any more changes, a maintainer will integrate
-     the change into our private repository, and it will get pushed out to the
-     public repository after some time.
+*   Include unit tests when you contribute new features, as they help to a)
+    prove that your code works correctly, and b) guard against future breaking
+    changes to lower the maintenance cost.
+*   Bug fixes also generally require unit tests, because the presence of bugs
+    usually indicates insufficient test coverage.
+*   When you contribute a new feature to Cobalt, the maintenance burden is
+    (by default) transferred to the Cobalbt team. This means that the benefit
+    of the contribution must be compared against the cost of maintaining the
+    feature.
+*   As every PR requires several CPU/GPU hours of CI testing, we discourage
+    submitting PRs to fix one typo, one warning,etc. We recommend fixing the
+    same issue at the file level at least (e.g.: fix all typos in a file, fix
+    all compiler warning in a file, etc.)
+
+#### License
+
+Include a license at the top of new files. Check existing files for license examples.
+
+#### Coding style
+
+Changes to Cobalt C++ code should conform to
+[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).
+Changes to Cobalt Python code should conform to
+[Google Python Style Guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md)
+
+
+#### Coding style
+
+Cobalt uses pre-commit to ensure good coding style. Create a python 3 virtual
+environment for working with Cobalt, then install `pre-commit` with:
+
+```bash
+$ pre-commit install -t post-checkout -t pre-commit -t pre-push --allow-missing-config
+```
+
+`pre-commit` will mostly run automatically, and can also be invoked manually.
+You can find documentation about it at https://pre-commit.com/.
+
+*   [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
+*   [Google Python Style Guide](https://google.github.io/styleguide/blob/gh-pages/pyguide.html)
+*   [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html)
+*   [Google JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html)
+*   [Google Shell Style Guide](https://google.github.io/styleguide/shellguide.html)
+*   [Google Objective-C Style Guide](https://google.github.io/styleguide/objcguide.html)
+
+#### Running unit tests
```

Comment:
Should we say anything about what to expect from CI runs ? E.g. that external contributor CI runs need to be approved, that style checks are enforced, and all checks are expected to be green before merging.

Also we don't, for now, set any expectations around review response rates or that the code will be accepted.

Reply:
I do mention that checks should pass in "Contributing code," but I made it a bit more pointed